### ISSUE TITLE
Issue in toFile method of Properties instance

### ISSUE THREAD

**Issue by acheronian** [vaind/properties.dart#4]
Changing or adding a property in the instantiated `Properties` entity and then executing it's `toFile` method results in weird file content:

`10997112115469711210584111107101110326132846983841010`

Code looks as followed:

```
void main() {

  var foo = AndroidProperties(mapsApiToken: 'TEST');
  foo.writeToProperties();

}

class AndroidProperties {

    AndroidProperties({
    this.mapsApiToken,
  });


    final String? mapsApiToken;

    void writeToProperties() {

      if(mapsApiToken.hasContent) {
        var mapPropertiesPath = 'android${Platform.isMacOS ? '\/' : '\\'}maps.properties';
        var mapProperties = Properties.fromFile(mapPropertiesPath);

        mapProperties.add('maps.apiToken', mapsApiToken!);
        mapProperties.toFile(mapPropertiesPath);
      }

    }

}
```

**Comment by vladchuk** [comment]
That API (toFile()) is unusual in that it uses events to compose file contents and codepoints to write them. It is not covered by tests and seems a bit of an experimentation. @vaind, perhaps it should be removed altogether?

There is now a counterpart (saveToFile()) that has been merged into main (not yet published in pub.dev), but you can use it locally.
It is using UTF-8 encoding, is covered by test and is being used.

One thing it doesn't do, which the "toFile" has a potential to do is preserve properties order and original comments. But that is rather a corner case, since, usually, existing configuration files are not manipulated programmatically. I am using it for the same purpose as you are: caching between sessions client tokens :)

**Comment by acheronian** [comment]
> That API (toFile()) is unusual in that it uses events to compose file contents and codepoints to write them. It is not covered by tests and seems a bit of an experimentation. @vaind, perhaps it should be removed altogether?
> 
> There is now a counterpart (saveToFile()) that has been merged into main (not yet published in pub.dev), but you can use it locally. It is using UTF-8 encoding, is covered by test and is being used.
> 
> One thing it doesn't do, which the "toFile" has a potential to do is preserve properties order and original comments. But that is rather a corner case, since, usually, existing configuration files are not manipulated programmatically. I am using it for the same purpose as you are: caching between sessions client tokens :)

Thank you, switching to the main branch and using saveToFile fixed the above issue for me.

Maybe make toFile a private or protected fun until it is covered by tests, but would open a different issue for this.

**Comment by vaind** [comment]
I've released 2.1.0 with the changes
